dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/BaseControls/Auxiliaries/UfMeasurement.mo

```diff
@@ -0,0 +1,84 @@
+within Dynawo.Electrical.Controls.IEC.BaseControls.Auxiliaries;
+
+/*
+* Copyright (c) 2024, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+model UfMeasurement "Measurement module for grid protection (IEC N°61400-27-1:2015)"
+
+  //Nominal parameters
+  parameter Types.Time tS "Integration time step in s";
+
+  //Uf measurement parameters
+  parameter Types.AngularVelocityPu DfMaxPu "Maximum frequency ramp rate in pu/s (base omegaNom)" annotation(
+    Dialog(tab = "UfMeasurement"));
+  parameter Types.Time tfFilt "Filter time constant for frequency measurement in s" annotation(
+    Dialog(tab = "UfMeasurement"));
+  parameter Types.Time tUFilt "Filter time constant for voltage measurement in s" annotation(
+    Dialog(tab = "UfMeasurement"));
+
+  //Input variables
+  Modelica.Blocks.Interfaces.RealInput omegaRefPu(start = Dynawo.Electrical.SystemBase.omegaRef0Pu) "Grid angular frequency in pu (base omegaNom)" annotation(
+    Placement(visible = true, transformation(origin = {-120, 0}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-142, -790}, extent = {{-42, -42}, {42, 42}}, rotation = 0)));
+  Modelica.ComplexBlocks.Interfaces.ComplexInput uWTPu(re(start = u0Pu.re), im(start = u0Pu.im)) "Complex voltage at grid terminal in pu (base UNom)" annotation(
+    Placement(visible = true, transformation(origin = {-120, 48}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-142, 814}, extent = {{-42, -42}, {42, 42}}, rotation = 0)));
+
+  //Output variables
+  Modelica.Blocks.Interfaces.RealOutput omegaGenPu(start = Dynawo.Electrical.SystemBase.omegaRef0Pu) "Filtered grid angular frequency in pu (base omegaNom)" annotation(
+    Placement(visible = true, transformation(origin = {160, 34}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {142, -790}, extent = {{-42, -42}, {42, 42}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealOutput UWTFiltPu(start = U0Pu) "Filtered voltage amplitude at grid terminal in pu (base UNom)" annotation(
+    Placement(visible = true, transformation(origin = {160, 74}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {143, 811}, extent = {{-43, -43}, {43, 43}}, rotation = 0)));
+
+  Modelica.ComplexBlocks.ComplexMath.ComplexToPolar complexToPolar annotation(
+    Placement(visible = true, transformation(origin = {-72, 46}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Continuous.Derivative derivative(k = 1 / SystemBase.omegaNom, x_start = UPhase0, y_start = 0) annotation(
```

Comment:
Just for information, what is the objective of doing this ? Why not just putting T = 0 ?

Reply:
I think it's just so it can be faster than the firstOrder (we put a value proportional to tfFilt instead of the default value 0.01).
T cannot be 0 because in the Derivative block it is in the denominator